### PR TITLE
meson: clear flags used for build

### PR DESCRIPTION
**Pull request by kasper93**

### CODE REVIEW

File: video/out/opengl/context_x11egl.c [hunk 1]
```diff
@@ -18,10 +18,14 @@
 #include <assert.h>
 
 #include <X11/Xlib.h>
-#include <X11/extensions/Xpresent.h>
```
Comment (Dudemanguy) [inline video/out/opengl/context_x11egl.c:21]: This header doesn't need to be here. Not sure why I put it here.

File: video/out/x11_common.c [hunk 2]
```diff
@@ -37,9 +37,13 @@
...
 #include <X11/extensions/dpms.h>
 #include <X11/extensions/shape.h>
-#include <X11/extensions/Xpresent.h>
 #include <X11/extensions/Xrandr.h>
 
+#ifndef PRESENT_FUTURE_VERSION
+#define PRESENT_FUTURE_VERSION 0
+#endif
+#include <X11/extensions/Xpresent.h>
+
```
[the remark this answers is not in the document]
Comment (kasper93) [reply video/out/x11_common.c:46]: See here https://github.com/kasper93/mpv/actions/runs/9374105487/job/25809404735

```
 In file included from ../video/out/x11_common.c:40:
In file included from /usr/local/include/X11/extensions/Xpresent.h:27:
/usr/local/include/X11/extensions/presenttokens.h:77:5: error: 'PRESENT_FUTURE_VERSION' is not defined, evaluates to 0 [-Werror,-Wundef]
#if PRESENT_FUTURE_VERSION
    ^
/usr/local/include/X11/extensions/presenttokens.h:85:5: error: 'PRESENT_FUTURE_VERSION' is not defined, evaluates to 0 [-Werror,-Wundef]
#if PRESENT_FUTURE_VERSION
    ^
/usr/local/include/X11/extensions/presenttokens.h:89:5: error: 'PRESENT_FUTURE_VERSION' is not defined, evaluates to 0 [-Werror,-Wundef]
#if PRESENT_FUTURE_VERSION
    ^
In file included from ../video/out/x11_common.c:40:
/usr/local/include/X11/extensions/Xpresent.h:109:5: error: 'PRESENT_FUTURE_VERSION' is not defined, evaluates to 0 [-Werror,-Wundef]
#if PRESENT_FUTURE_VERSION
    ^
4 errors generated.
```

[the remark this answers is not in the document]
Comment (Dudemanguy) [reply video/out/x11_common.c:46]: Huh weird. I had no idea this define was used in the headers here.

File: video/out/opengl/context_x11egl.c (hunk 1 above)
Comment (kasper93) [reply video/out/opengl/context_x11egl.c:21]: removed

File: meson.build [hunk 3]
```diff
@@ -266,43 +266,30 @@ cc = meson.get_compiler('c')
...
+              '-Wno-sign-compare',
+              '-Wno-switch',
+              '-Wno-unused-parameter',
               '-fno-math-errno']
 
 flags += cc.get_supported_arguments(test_flags)
 
-if cc.has_multi_arguments('-Wformat', '-Werror=format-security')
-    flags += ['-Wformat', '-Werror=format-security']
-endif
```
Comment (Dudemanguy) [inline meson.build:288]: `'-Werror=format-security'` requires the `has_multi_arguments` test. It doesn't work with `get_supported_arguments` which only supports one.

Comment (kasper93) [reply meson.build:288]: Just GCC things. I see that it errors out when it is used alone. Will update.

**Review by Dudemanguy — APPROVED** [review]

**Pull request merged**